File: Singularity Trainer/Assets/Src/Actions/ShootAction.cs
using Scripts.Modules.Interfaces;
using Scripts.Modules;

namespace Actions
{
    public class ShootAction : Action
    {
        public ShootAction(Module parentModule) : base(parentModule)
        {
            options = 1;
        }

        public override void Act(List<bool> actions)
        {
            if (Module is IShootable && actions[0] == true)
            {
                (Module as IShootable).Shoot();
            }
        }
    }
}

File: cpp/src/training/actions/shoot_action.cpp
#include <vector>

#include "training/actions/shoot_action.h"
#include "training/modules/interfaces/ishootable.h"

namespace SingularityTrainer
{
ShootAction::ShootAction(IShootable &module) : module(module)
{
    flag_count = 1;
}
ShootAction::~ShootAction() {}

void ShootAction::act(std::vector<int> flags)
{
    if (flags[0] == 1)
    {
        module.shoot();
    }
}
}